File: Binary Tree Level Order Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void helper(vector<TreeNode *> levelNodes, vector<vector<int> > &res) {
        vector<TreeNode *> nextlevelNodes;
        vector<int> vals;
        for (int i = 0; i < levelNodes.size(); i++) {
            TreeNode *tmp = levelNodes[i];
            if (tmp->left) {
                nextlevelNodes.push_back(tmp->left);
            }
            if (tmp->right) {
                nextlevelNodes.push_back(tmp->right);
            }
            vals.push_back(tmp->val);
        }
        
        res.push_back(vals);
        
        if (nextlevelNodes.size() != 0) {
            helper(nextlevelNodes, res);
        }
    }
    vector<vector<int> > levelOrder(TreeNode *root) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<vector<int> > res;
        if (root == NULL) return res;
        vector<TreeNode *> nodes;
        nodes.push_back(root);
        helper(nodes, res);
        return res;
    }
};

File: Binary Tree Level Order Traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        var result = new List<IList<int>>();
        if (root == null) {
            return result;
        }
        
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        while (queue.Count > 0) {
            var size = queue.Count;
            var values = new List<int>();
            for (var i = 0; i < size; i++) {
                var cur = queue.Dequeue();
                values.Add(cur.val);
                if (cur.left != null) {
                    queue.Enqueue(cur.left);
                }
                if (cur.right != null) {
                    queue.Enqueue(cur.right);
                }
            }
            result.Add(values);
        }
        
        return result;
    }
}
